ZichanYang324/BF-Employee-HR
# BGP-44 Housing Page Frontend - HR side Angular

# Description
Implemented Housing, Report, and Comment pages for the HR frontend and updated backend. 

Modified GET to POST HTTP request method for `getAllBasicHouseInfoForHR`, `getHouseSummaryForHR`, `getComment`, enabling profileId embedded in the body.

**Housing**
- View housing information including address and roommates
- Add a house
- Delete a house

**Facility Report**
- View all reports by page

**Comment**
- View all comments (User and HR)
- Create a comment
- Edit a comment


## Related PRS (if any):

BGP#12, BGP#18


## Main changes explained:
- Added housing component
- Added report component
- Added comment component
- Added Housing, Report, Comment Services
- Modify Server Router, change get method from 'get' to 'post'

## How to test:

1. check into the current branch
2. Run `npm install` for both frontend and backend `npm run dev`(Server) ,`ng serve`(frontend)
3. navigate to '/housing' to test the housing info, create new a house, delete a house, and access to the report and comment


- '/housing' - housing page for the HR
- '/house-summary/:housingID?page=1 for summary report

## Screenshots or videos of changes or expecting result:
<img width="1440" alt="Screen Shot 2024-03-03 at 9 56 26 AM" src="https://github.com/ZichanYang324/BF-Employee-HR/assets/36239681/69bdf6ac-3aa1-4af6-ba39-51ef22a4eca4">
<img width="1440" alt="Screen Shot 2024-03-03 at 9 57 00 AM" src="https://github.com/ZichanYang324/BF-Employee-HR/assets/36239681/b5f4b2f8-10a9-43e0-8945-1b0a64435fab">



## Note:
1. Currently, the userId is hardcoded in the service file  and will be replaced when the profileId is encoded in JWT.
